### PR TITLE
this allows gcc4.9.3 to build on OSX10

### PR DESCRIPTION
**Pull request by gartung**
Thought you might be interested.

### CODE REVIEW

File: lib/spack/env/cc [hunk 1]
```diff
@@ -108,6 +108,12 @@ case "$command" in
         ;;
     ld)
         mode=ld
+        for arg in "$@"; do
+            if [ "$arg" = -S ]; then
```
Comment (eschnett) [inline lib/spack/env/cc:112]: This catches the case where `ld` is called with an `-S` option. I don't think this should happen -- the `-S` option is for the compiler driver `cc`.

There apparently is an `-S` option for `ld`; it means the opposite of `-g`, i.e. remove (Strip) all debug information. This has nothing to do with the mode (the mode should still be `ld`), nor with how `-rpath` should be handled.

Comment (gartung) [inline lib/spack/env/cc:112]: Somehow when using the --with-build-option=bootstrap-debug "ld -S -r" is invoked and spack intercepts this. Here is an excerpt from the spack debug log when spack incorrectly added the -rpath. The "out" command fails.

in: ld  -S -r -no_uuid /private/var/folders/hm/7pxrf4vx4_g4f_nkfrcwqq8m0000gp/T/gartung/spack-stage/spack-stage-PZv9zT/spack-build/stage2-x86_64-apple-darwin14.5.0/libstdc++-v3/src/compatibility.o -o /private/var/folders/hm/7pxrf4vx4_g4f_nkfrcwqq8m0000gp/T/gartung/spack-stage/spack-stage-PZv9zT/spack-build/stage2-x86_64-apple-darwin14.5.0/libstdc++-v3/src/compatibility.o.stripped

out:    as  ld -L/Users/gartung/spack/opt/spack/darwin-x86_64/clang-7.0.2-apple/mpfr-3.1.4-bvgor6plszncasuc2wtpo6uuuoq23wxt/lib -rpath /Users/gartung/spack/opt/spack/darwin-x86_64/clang-7.0.2-apple/mpfr-3.1.4-bvgor6plszncasuc2wtpo6uuuoq23wxt/lib -L/Users/gartung/spack/opt/spack/darwin-x86_64/clang-7.0.2-apple/mpc-1.0.3-jhigwld6okg5lvlvzugnm43nchbgzq2m/lib -rpath /Users/gartung/spack/opt/spack/darwin-x86_64/clang-7.0.2-apple/mpc-1.0.3-jhigwld6okg5lvlvzugnm43nchbgzq2m/lib -L/Users/gartung/spack/opt/spack/darwin-x86_64/clang-7.0.2-apple/libiconv-1.14-6334hosj4o4mzeij5wgzebuzfywfp74a/lib -rpath /Users/gartung/spack/opt/spack/darwin-x86_64/clang-7.0.2-apple/libiconv-1.14-6334hosj4o4mzeij5wgzebuzfywfp74a/lib -L/Users/gartung/spack/opt/spack/darwin-x86_64/clang-7.0.2-apple/gmp-6.1.0-kmnkwalnopisi4yxpk6b5fspre3cqlps/lib -rpath /Users/gartung/spack/opt/spack/darwin-x86_64/clang-7.0.2-apple/gmp-6.1.0-kmnkwalnopisi4yxpk6b5fspre3cqlps/lib -S -r -no_uuid /private/var/folders/hm/7pxrf4vx4_g4f_nkfrcwqq8m0000gp/T/gartung/spack-stage/spack-stage-PZv9zT/spack-build/stage2-x86_64-apple-darwin14.5.0/libstdc++-v3/src/compatibility.o -o /private/var/folders/hm/7pxrf4vx4_g4f_nkfrcwqq8m0000gp/T/gartung/spack-stage/spack-stage-PZv9zT/spack-build/stage2-x86_64-apple-darwin14.5.0/libstdc++-v3/src/compatibility.o.stripped

Comment (eschnett) [inline lib/spack/env/cc:112]: Why is there an `as` in front of the `ld` in the output? That should not be there.

Comment (gartung) [inline lib/spack/env/cc:112]: The output was from an earlier modification I made and then abandoned because it did not fix the problem.

**Pull request closed without merging**